src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,258 @@
+            // Initialize the default modes
+            _measurementMode = MeasurementMode.PowerDown;
+            _outputBitMode = OutputBitMode.Output14bit;
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            Thread.Sleep(100);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns></returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias => _magnetometerBias;
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer()
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
+            ReadByteArray(Register.ASAX, rawData);
+            calib.X = (float)((rawData[0] - 128) / 256.0 + 1.0);
+            calib.Y = (float)((rawData[1] - 128) / 256.0 + 1.0);
+            calib.Z = (float)((rawData[2] - 128) / 256.0 + 1.0);
+            MeasurementMode = MeasurementMode.PowerDown;
+            MeasurementMode = oldPower;
+
+            // Now calculate the bias
+            // Store old mode to restore after
+            var oldMode = MeasurementMode;
+            Vector3 minbias = new Vector3();
+            Vector3 maxbias = new Vector3();
+            int numberMeasurements = 1500;
+
+            // Setup the 100Hz continous mode
+            MeasurementMode = MeasurementMode.ContinousMeasurement100Hz;
+            for (int reading = 0; reading < numberMeasurements; reading++)
+            {
+                var bias = ReadMagnetometer(true);
+                minbias.X = bias.X < minbias.X ? bias.X : minbias.X;
+                minbias.Y = bias.Y < minbias.Y ? bias.Y : minbias.Y;
+                minbias.Z = bias.Z < minbias.Z ? bias.Z : minbias.Z;
+                maxbias.X = bias.X > maxbias.X ? bias.X : maxbias.X;
+                maxbias.Y = bias.Y > maxbias.Y ? bias.Y : maxbias.Y;
+                maxbias.Z = bias.Z > maxbias.Z ? bias.Z : maxbias.Z;
+                Thread.Sleep(12);
```

Comment:
please add a comment why sleep is needed (and point to spec)

Reply:
(should the Read do the sleep if this is related with that?)

Reply:
at 100Hz, you need to wait at least 10 milliseconds to have a new reading. 12 is a safe way to have it. If you read faster, you won't get a new one. This is just to make sure we have data rather than reading the resgister all the time to get new data. I will adjust to 10 and leave a comment.

Reply:
@Ellerbach as per later feedback you should have it directly on Read or create WaitForXyz method